src/Doctrine/DocumentRegenerator.php

```diff
@@ -0,0 +1,236 @@
+
+            $manipulator = $this->createClassManipulator($classPath);
+            $operations[$classPath] = $manipulator;
+
+            $embeddedClasses = [];
+
+            foreach ($classMetadata->getEmbeddedFieldsMappings() as $fieldName => $mapping) {
+                if (str_contains($fieldName, '.')) {
+                    continue;
+                }
+
+                $className = $mapping['targetDocument'];
+
+                $embeddedClasses[$fieldName] = $this->getPathOfClass($className);
+
+                $operations[$embeddedClasses[$fieldName]] = $this->createClassManipulator($embeddedClasses[$fieldName]);
+
+                if (!\in_array($fieldName, $mappedFields)) {
+                    continue;
+                }
+
+                $manipulator->addEmbeddedDocument($fieldName, $className);
+            }
+
+            foreach ($classMetadata->fieldMappings as $fieldName => $mapping) {
+                // skip embedded fields
+                if (str_contains($fieldName, '.')) {
+                    [$fieldName, $embeddedFiledName] = explode('.', $fieldName);
+
+                    $operations[$embeddedClasses[$fieldName]]->addDocumentField($embeddedFiledName, $mapping);
+
+                    continue;
+                }
+
+                if (!\in_array($fieldName, $mappedFields)) {
+                    continue;
+                }
+
+                $manipulator->addDocumentField($fieldName, $mapping);
+            }
+
+            $getIsNullable = function (array $mapping) {
+                return true;
+            };
+
+            foreach ($classMetadata->associationMappings as $fieldName => $mapping) {
+                if (!\in_array($fieldName, $mappedFields)) {
+                    continue;
+                }
+
+                switch ($mapping['type']) {
+                    case ClassMetadata::ONE:
+                        $relation = (new ReferenceOne(
+                            propertyName: $mapping['fieldName'],
+                            targetDocument: $mapping['targetDocument'],
+                            targetPropertyName: @$mapping['inversedBy'],
+                            mapInverseRelation: null !== @$mapping['inversedBy'],
+                            isOwning: true,
+                            isNullable: $getIsNullable($mapping),
+                        ));
+
+                        $manipulator->addManyToOneReference($relation);
+
+                        break;
+                    case ClassMetadata::MANY:
+                        $relation = (new ReferenceMany(
+                            propertyName: $mapping['fieldName'],
+                            targetDocument: $mapping['targetDocument'],
+                            targetPropertyName: $mapping['mappedBy'],
+                            orphanRemoval: $mapping['orphanRemoval'],
+                        ));
+
+                        $manipulator->addOneToManyReference($relation);
+
+                        break;
```

Comment:
Associations in ODM come in two variants: references and embedded documents. You assume that any association is a reference, which is incorrect. You need to differentiate on `$mapping['embedded']` to distinguish between references and embedded documents.